src/QsCompiler/DataStructures/RuntimeCapability.fs

```diff
@@ -3,76 +3,139 @@
 
 namespace Microsoft.Quantum.QsCompiler
 
-#nowarn "44" // RuntimeCapabilities is deprecated.
-
-open Microsoft.Quantum.QsCompiler.DataTypes
-open Microsoft.Quantum.QsCompiler.ReservedKeywords.AssemblyConstants
 open System
-open System.Runtime.CompilerServices
 
-/// The runtime capabilities supported by a quantum processor.
+[<CustomComparison>]
+[<CustomEquality>]
+type CaseInsensitive =
+    | CaseInsensitive of string
+
+    member ci.Original =
+        let (CaseInsensitive s) = ci
+        s
+
+    override ci.Equals obj = (ci :> IComparable).CompareTo obj = 0
+
+    override ci.GetHashCode() =
+        StringComparer.InvariantCultureIgnoreCase.GetHashCode ci.Original
+
+    interface IComparable with
+        member ci.CompareTo obj =
+            match obj with
+            | :? CaseInsensitive as other ->
+                String.Compare(ci.Original, other.Original, StringComparison.InvariantCultureIgnoreCase)
+            | _ -> ArgumentException("Type mismatch.", nameof obj) |> raise
+
+module CaseInsensitive =
+    let original (CaseInsensitive s) = s
+
+type ResultOpacity =
+    | Opaque
+    | Controlled
```

Comment:
I think controlled result opacity correlates with what is possible to lift into the [classical control operations](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.classicalcontrol)? That was my idea for the name.

Reply:
For example here you see that `ConvertClassicalControl` is true if the result opacity is `Controlled`:

https://github.com/microsoft/qsharp-compiler/blob/e81e3845a3fef4ae52624d763b42b6cef42d6a83/src/QsCompiler/Compiler/CompilationLoader.cs#L205-L206

I like the symmetry and would prefer to avoid using a different word.

Reply:
It does indeed. I am fine keeping it as is.